commands/audit/sca/yarn/yarn.go

```diff
@@ -136,19 +137,23 @@ func configureYarnResolutionServerAndRunInstall(params utils.AuditParams, curWd,
 	return runYarnInstallAccordingToVersion(curWd, yarnExecPath, params.InstallCommandArgs())
 }
 
-func isInstallRequired(currentDir string, installCommandArgs []string) (installRequired bool, err error) {
+// We verify the project's installation status by examining the presence of the yarn.lock file and the presence of an installation command provided by the user.
+// If install command was provided - we install
+// If yarn.lock is missing, we should install unless the user has explicitly disabled auto-install. In this case we return an error
+// Notice!: If alterations are made manually in the package.json file, it necessitates a manual update to the yarn.lock file as well.
+func isInstallRequired(currentDir string, installCommandArgs []string, skipAutoInstall bool) (installRequired bool, err error) {
 	yarnLockExits, err := fileutils.IsFileExists(filepath.Join(currentDir, yarn.YarnLockFileName), false)
 	if err != nil {
 		err = fmt.Errorf("failed to check the existence of '%s' file: %s", filepath.Join(currentDir, yarn.YarnLockFileName), err.Error())
 		return
 	}
 
-	// We verify the project's installation status by examining the presence of the yarn.lock file and the presence of an installation command provided by the user.
-	// Notice!: If alterations are made manually in the package.json file, it necessitates a manual update to the yarn.lock file as well.
-	if len(installCommandArgs) > 0 || !yarnLockExits {
-		installRequired = true
+	if len(installCommandArgs) > 0 {
+		return true, nil
```

Comment:
```suggestion
	if len(installCommandArgs) > 0 {
	        if skipAutoInstall {
	               log.Warn("Conflict: Requested to skip install but provided an install command")
	        }
		return true, nil
```

maybe skip should take priority always? even if cmd is provided? I think we should consult product

Reply:
this is what was approved by product